src/backend/joanie/tests/test_helpers.py

```diff
@@ -0,0 +1,196 @@
+"""Joanie core helpers tests suite"""
+from datetime import timedelta
+from unittest import mock
+
+from django.test.testcases import TestCase
+from django.utils import timezone
+
+from joanie.core import enums, factories, helpers, models
+from joanie.lms_handler.backends.dummy import DummyLMSBackend
+
+
+class HelpersTestCase(TestCase):
+    """Joanie core helpers tests case"""
+
+    def test_helpers_generate_certificate_for_order_needs_graded_courses(self):
+        """
+        If the order relies on a certifying product which does not contain
+        graded courses any certificate should be generated.
+        """
+        # Create a certifying product with one order eligible for certification
+        course_run = factories.CourseRunFactory(
+            enrollment_end=timezone.now() + timedelta(hours=1),
+            enrollment_start=timezone.now() - timedelta(hours=1),
+            is_gradable=True,
+            start=timezone.now() - timedelta(hours=1),
+        )
+        product = factories.ProductFactory(
+            price="0.00",
+            type=enums.PRODUCT_TYPE_CREDENTIAL,
+            target_courses=[course_run.course],
+        )
+        # Mark the only product's course as non graded
+        course_run.course.product_relations.update(is_graded=False)
+        course = factories.CourseFactory(products=[product])
+        order = factories.OrderFactory(product=product, course=course)
+        certificate = models.Certificate.objects.filter(order=order)
```

Comment:
I think it shouldn't be called certificate when it's a query. Same for other occurences in other test files.

Reply:
I renamed it `certificate_qs`